File: 229.Majority Element II/src/Solution.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> list = new ArrayList<Integer>();
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; i++) {
        	if (map.get(nums[i]) == null) {
        		map.put(nums[i], 1);
        	}
        	else {
        		map.put(nums[i], map.get(nums[i] + 1));
        	}
        }
        
        for (Integer in : map.keySet()) {
        	if (map.get(in) > nums.length/3)
        		list.add(in);
        }
        return list;
    }
}

File: 229.Majority Element II/Solution.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = []
        temp = {}
        for i in range(len(nums)):
            if nums[i] in temp.keys():
                temp[nums[i]] += 1
            else:
                temp[nums[i]] = 1
        for k in temp.keys():
            if temp[k] > len(nums) / 3:
                res.append(k)
        return res


        # tmp = {}
        # res = []
        # for n in list(set(nums)):
        #     tmp[n] = nums.count(n)
        # for k,v in tmp.iteritems():
        #     if v > len(nums) / 3:
        #         res.append(k)
        # return res